### ISSUE TITLE
Loading pickled models cause NotFittedError

### ISSUE THREAD

**Issue by lw3259111** [flennerhag/mlens#77]
Thanks yours model ,this is great idea.
I use the model in mulit layer, and use pickle dump to store the train model,but when load model which have beed stored some error is found .
`Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/python3.5/lib/python3.5/site-packages/mlens/ensemble/base.py", line 615, in predict
    return self._backend.predict(X, **kwargs)
  File "/usr/python3.5/lib/python3.5/site-packages/mlens/ensemble/base.py", line 206, in predict
    out = self._predict(X, 'predict', **kwargs)
  File "/usr/python3.5/lib/python3.5/site-packages/mlens/ensemble/base.py", line 266, in _predict
    out = manager.stack(self, job, X, return_preds=r, **kwargs)
  File "/usr/python3.5/lib/python3.5/site-packages/mlens/parallel/backend.py", line 655, in stack
    return self.process(caller=caller, out=out, **kwargs)
  File "/usr/python3.5/lib/python3.5/site-packages/mlens/parallel/backend.py", line 700, in process
    self._partial_process(task, parallel, **kwargs)
  File "/usr/python3.5/lib/python3.5/site-packages/mlens/parallel/backend.py", line 721, in _partial_process
    task(self.job.args(**kwargs), parallel=parallel)
  File "/usr/python3.5/lib/python3.5/site-packages/mlens/parallel/layer.py", line 123, in __call__
    "Layer instance (%s) not fitted." % self.name)
mlens.utils.exceptions.NotFittedError: Layer instance (layer-1) not fitted.
`

so how can i store and use my model of trained?

**Comment by flennerhag** [comment]
Hi @lw3259111. 

This should work – could you provide info on which platform your on (Win/Mac/Linux) and which Python version your using. Also helpful if you could provide a minimum example that causes the error.

**Comment by lw3259111** [comment]
@flennerhag 
Thanks for your reply, i use mlens in python3 and linux platform,.
`ensemble = SequentialEnsemble(scorer=rmse)
    ensemble.add('stack',list(model.values()))
    ensemble.add('blend',list(model.values()))
    ensemble.add('subsemble',list(model.values()))
    meta = "ls"
    ensemble.add_meta(model[meta])
    ensemble.fit(X_train_train,y_train_train)
    pickle.dump(ensemble, open("ensemble_zero_to_three.model", "wb"))
`
this is my core code for mlens.

**Comment by lw3259111** [comment]
@flennerhag i test the script in linux with python2.7.12. I had test the sklearn joblib, The problem happened again, so, how can i solved the problem.
sklearn joblib code:
`
ensemble = SequentialEnsemble(scorer=rmse)
ensemble.add('stack',list(model.values()))
ensemble.add('blend',list(model.values()))
ensemble.add('subsemble',list(model.values()))
meta = "ls"
ensemble.add_meta(model[meta])
ensemble.fit(X_train,y_train)
preds1 = ensemble.predict(X)
predict = ensemble.predict(predictData)
try:
    filename = 'ensemble_zero_to_three.joblib'
    joblib.dump(ensemble, filename)
except:
    pass
`

**Comment by lw3259111** [comment]
@flennerhag When my model combines with xgboost  and lightgbm, the pickle not work.so, can you help me ?

**Comment by flennerhag** [comment]
@lw3259111 the problem was that XGBRegressor has a parameter that is ``nan``, which is not well behaved when you check parameter equivalence with ``==``. This has been fixed as of PR #79 and will be part of the next version update (0.2.2). 

Until then, install the repo's ``master`` branch version.

You can test that it works with the below MWE 

```python
from xgboost import XGBRegressor
from mlens.testing.dummy import return_pickled
from mlens.ensemble import SuperLearner
import numpy as np

X, y = np.random.rand(100, 10), np.random.rand(100)
ens = SuperLearner()
ens.add([XGBRegressor()])
 
ens.fit(X, y)
ens.predict(X)
ens = return_pickled(ens)
ens.predict(X)
```
